Source/CesiumRuntime/Private/CesiumTextureUtility.cpp

```diff
@@ -1,20 +1,429 @@
+      this->TextureRHI = pAsyncTexture->rhiTextureRef;
+      pAsyncTexture->rhiTextureRef.SafeRelease();
+    }
+  }
+
+  virtual ~FCesiumTextureResource() {
+    check(this->_pTexture != nullptr);
+    this->_pTexture->SetResource(nullptr);
+  }
+
+  uint32 GetSizeX() const override { return this->_width; }
+
+  uint32 GetSizeY() const override { return this->_height; }
+
+  virtual void InitRHI() override {
+    FSamplerStateInitializerRHI samplerStateInitializer(
+        this->_filter,
+        this->_addressX,
+        this->_addressY,
+        AM_Wrap,
+        0.0f,
+        4,
+        0.0f,
+        FLT_MAX);
+    this->SamplerStateRHI = GetOrCreateSamplerState(samplerStateInitializer);
+
+    FSamplerStateInitializerRHI deferredSamplerStateInitializer(
+        this->_filter,
+        this->_addressX,
+        this->_addressY,
+        AM_Wrap,
+        0.0f,
+        1,
+        0.0f,
+        2.0f);
+    this->DeferredPassSamplerStateRHI =
+        GetOrCreateSamplerState(deferredSamplerStateInitializer);
+
+    if (!this->TextureRHI) {
+      // Asynchronous RHI texture creation was not available. So create it now
+      // directly from the in-memory cesium mips.
+      CesiumGltf::ImageCesium* pImage =
+          std::visit(GetImageFromSource{}, this->_textureSource);
+
+      check(pImage != nullptr);
+
+      // Wrap mip0 as a bulk data source.
+      FCesiumTextureData bulkData(*pImage);
+
+      FRHIResourceCreateInfo createInfo{};
+      createInfo.BulkData = &bulkData;
+      createInfo.ExtData = _platformExtData;
+
+      ETextureCreateFlags textureFlags = TexCreate_ShaderResource;
+
+      if (this->bSRGB) {
+        textureFlags |= TexCreate_SRGB;
+      }
+
+      uint32 mipCount =
+          FMath::Max(1, static_cast<int32>(pImage->mipPositions.size()));
+
+      FTexture2DRHIRef rhiTexture;
+
+      // Copies over mip0, allocates the rest of the mips if needed.
+
+      // RHICreateTexture2D can actually copy over all the mips in one shot,
+      // but it expects a particular memory layout. Might be worth configuring
+      // Cesium Native's mip-map generation to obey a standard memory layout.
+      rhiTexture = RHICreateTexture2D(
+          this->_width,
+          this->_height,
+          this->_format,
+          mipCount,
+          1,
+          textureFlags,
+          createInfo);
+
+      // Copies over rest of the mips
+      for (uint32 i = 1; i < mipCount; ++i) {
+        uint32 DestPitch;
+        void* pDestination =
+            RHILockTexture2D(rhiTexture, i, RLM_WriteOnly, DestPitch, false);
+        CopyMip(pDestination, DestPitch, _format, *pImage, i);
+        RHIUnlockTexture2D(rhiTexture, i, false);
+      }
+
+      this->TextureRHI = rhiTexture;
+      rhiTexture.SafeRelease();
+    }
+
+    RHIUpdateTextureReference(TextureReferenceRHI, this->TextureRHI);
+
+    // TODO: investigate if we can / should do this!
+    // related to: https://github.com/CesiumGS/cesium-unreal/issues/980
```

Comment:
I investigated this again, and it doesn't look to make much of a difference. Deferred delete just means it waits for a render thread - rhi thread sync so it can safely delete the rhi object (from what I can tell). Explicitly not deferring delete might a) be unsafe, or b) force an unnecessary sync-up.

I can't be too confident about my current understanding of this, but I'm almost positive it's not related to the GC pressure we're seeing now - so I removed the comment! (There's no other issue that we could or need to solve by using DoNoDeferDelete anyways)